text/0670-aws-applicationsignals-enablement-l2.md

```diff
@@ -0,0 +1,633 @@
+such as add-on on EKS, operator on native Kubernetes, and vended layer on Lambda. The pain point is most prominent on ECS and EC2.
+
+### Why should we *not* do this?
+
+We should not using this feature if granular control over the TaskDefinition configuration is required. While the automation simplifies setup, it may
+not be suitable for cases where manually customizing specific fields or maintaining precise control over the TaskDefinition settings is needed. In
+such cases, stick with the existing L1 ECS TaskDefinition construct for manual configuration.
+
+### What is the technical solution (design) of this feature?
+
+#### 1. Current enablement process
+
+At high level, the customers are required to execute the following steps to onboard Application Signals.
+
+1. Add additional CloudWatch agent permission to the Task Role.
+2. Instrument ECS [TaskDefinition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html) with 5 steps.
+    1. Configuring the bind mount volume to provide a volume for auto-instrumentation.
+    2. Adding a new non-essential CloudWatch agent sidecar.
+        1. *Not required when using CloudWatch agent as daemon on ECS on EC2.*
+    3. Adding a new non-essential `init` container with ADOT auto-instrumentation image.
+    4. Adding the required environment variables to the user application container.
+    5. Mounting the volume defined in \<2.i\> to the container created in \<2.iii\> and the container updated in \<2.iv\>.
+3. Deploy the application with the updated task definition.
+
+The final TaskDefinition will be provisioned as illustrated in the following diagram.
+
+```text
+    --------                                    -----------------
+--- | main | ---------------------------------- | shared volume |
+|   --------                                    -----------------
+|       | copy the auto-instrumentation agent from    |
+|       | init into main for data collection          |
+|   --------                                          |
+|   | init | ------------------------------------------
+|   --------
+|
+| the auto-instrumentation agent sends monitoring data to the CloudWatch Agent
+|
+|   -----------
+--- | cwagent |
+    -----------
+```
+
+The init container is a short-lived container that copies the auto-instrumentation agent to the main container for data collection through the shared
+volume. The auto-instrumentation agent will be hooked into the application process on start, generating monitoring data (mainly traces and metrics)
+and sending them to the CloudWatch Agent. The CloudWatch Agent will further process the data received and export to the CloudWatch backend service.
+
+#### 2. Automated solution
+
+The new feature introduces an L2 construct (ApplicationSignalsTaskDefinition) that automates step 2 (including the 5 sub-steps by automatically
```

Comment:
it seems you missed updating this part, as we agreed to automate the first 2 steps, not only step 2

Reply:
Updated